### PR TITLE
Feature/add agencies page

### PR DESCRIPTION
**Pull request by Aleksandra05sh**
Add Agencies Page (https://trello.com/c/AOs3ciQC/11-add-agencies-page)
image

- added a header to the page (style - created mixin)
- created a grid of agencies
- added  agencies info (logo, description, info button)

### CODE REVIEW

File: src/components/Agencies/Agencies.tsx [hunk 1]
```diff
@@ -1,3 +1,13 @@
+import { Container, Grid, Paper } from '@material-ui/core';
 import React from 'react';
 
-export const Agencies = () => <div>Agencies</div>;
+export const Agencies = () => (
+  <>
+    <h1>Agencies</h1>
```
Comment (vovkis13) [inline src/components/Agencies/Agencies.tsx:6]: title should be a separate component in /shared/components and can be used in other components

File: src/components/Agencies/Agencies.tsx [hunk 2]
```diff
@@ -1,3 +1,13 @@
...
+    <Paper>
+      <Container>
+        <Grid container></Grid>
+      </Container>
+    </Paper>
+  </>
+);
```
Comment (vovkis13) [inline src/components/Agencies/Agencies.tsx:13]: this components isn't used

Comment (topsart) [reply src/components/Agencies/Agencies.tsx:13]: she fixed that in the next commit

Comment (topsart) [reply src/components/Agencies/Agencies.tsx:13]: try to sign commits more informatively

File: src/components/AgenciesPage/types/Agencies.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+}
+
+export interface Agencies extends AgenciesName {
+  type: string;
+  administrator: string;
+  country_code: string;
+  launchers: string;
+  image: string;
+  spacecraft?: string;
+  [key: string]: string | undefined;
```
Comment (vovkis13) [inline src/components/AgenciesPage/types/Agencies.ts:12]: what's the purpose of this line?

File: src/components/AgenciesPage/AgenciesPage.tsx [hunk 4]
```diff
@@ -1,10 +1,12 @@
...
+
 import { AgenciesList } from './components';
 import './AgenciesPage.scss';
 
 export const AgenciesPage = () => (
-  <section className="agencies-page">
+  <Container maxWidth="xl">
     <h1 className="agencies-header">Agencies</h1>
     <AgenciesList />
-  </section>
```
Comment (topsart) [inline src/components/AgenciesPage/AgenciesPage.tsx]: Why did you remove the section tag here?

File: src/components/AgenciesPage/components/AgenciesList.tsx [hunk 5]
```diff
@@ -1,4 +1,14 @@
+import { Grid } from '@mui/material';
 import React from 'react';
 import { AgenciesItem } from './components';
 
-export const AgenciesList = () => <AgenciesItem />;
+const cards = [1, 2, 3, 4, 5, 6];
+export const AgenciesList = () => (
+  <Grid className="agencies-grid-container" container>
+    {cards.map(card => (
+      <Grid item key={card} xs={12} sm={6} md={4} lg={3}>
```
Comment (topsart) [inline src/components/AgenciesPage/components/AgenciesList.tsx:9]: we use the index as a key

File: src/components/AgenciesPage/types/Agencies.ts [hunk 6]
```diff
@@ -0,0 +1,19 @@
+interface AgenciesName {
+  name: string;
+}
+
+export interface Agencies extends AgenciesName {
```
Comment (vovkis13) [inline src/components/AgenciesPage/types/Agencies.ts:5]: you can add 'id' for further using in key prop

File: src/components/AgenciesPage/types/Agencies.ts (hunk 3 above)
Comment (Aleksandra05sh) [reply src/components/AgenciesPage/types/Agencies.ts]: https://www.typescriptlang.org/docs/handbook/2/objects.html#index-signatures

I use it because keys were not defined as strings

File: src/components/AgenciesPage/components/AgenciesList.tsx (hunk 5 above)
Comment (Aleksandra05sh) [reply src/components/AgenciesPage/components/AgenciesList.tsx]: agree, it was for test, now i use Id

File: src/components/AgenciesPage/components/components/AgenciesItem.tsx [hunk 7]
```diff
@@ -0,0 +1,15 @@
+import React from 'react';
+import { Card } from '@mui/material';
+import { AgenciesItemImage } from './components/AgenciesItemImage';
+import { AgenciesItemContent } from './components/AgenciesItemContent';
+import { AgenciesItemActions } from './components/AgenciesItemActions';
+import './AgenciesItem.scss';
+import type { AgenciesItemProps } from '../../types/Agencies';
+
+export const AgenciesItem: React.FC<AgenciesItemProps> = ({ agency }) => (
```
Comment (PietyE) [inline src/components/AgenciesPage/components/components/AgenciesItem.tsx:9]: Should we typing components as React.FC or it isnt necessary ?

File: src/components/AgenciesPage/types/Agencies.ts (hunk 3 above)
Comment (PietyE) [reply src/components/AgenciesPage/types/Agencies.ts]: We can use one more variant for resolve this error , i had this error recently 
**My variant example:** 
interface Object extends Record<string, any>{
  name?: string;
  age?: number;
}

const object: Object = {
  name: 'Dima',
  age: 18,
};

Comment (Aleksandra05sh) [reply src/components/AgenciesPage/types/Agencies.ts]: thank you 
but i'm not sure, that we cat use any

File: src/components/AgenciesPage/components/components/AgenciesItem.tsx [hunk 8]
```diff
@@ -0,0 +1,15 @@
+import React from 'react';
+import { Card } from '@mui/material';
+import { AgenciesItemImage } from './components/AgenciesItemImage';
+import { AgenciesItemContent } from './components/AgenciesItemContent';
+import { AgenciesItemActions } from './components/AgenciesItemActions';
+import './AgenciesItem.scss';
```
[the remark this answers is not in the document]
Comment (Aleksandra05sh) [reply src/components/AgenciesPage/components/components/AgenciesItem.tsx:6]: agree with you 
I don't sort imports yet

File: src/components/AgenciesPage/components/components/AgenciesItem.tsx (hunk 7 above)
Comment (Aleksandra05sh) [reply src/components/AgenciesPage/components/components/AgenciesItem.tsx:9]: we cat use another approach: ({ agency : AgenciesItemProps})
But in this case, our type should  contain children's prop if we are going to use it 
that's why I prefer my option and consider it as more readable

File: src/components/AgenciesPage/components/components/components/AgenciesItemContent/AgenciesItemContent.tsx [hunk 9]
```diff
@@ -0,0 +1,27 @@
+import React from 'react';
+import { CardContent, Typography } from '@mui/material';
+import { agenciesData } from '../../../../mocks/agenciesData';
+import { SpecificationsTable } from './components';
+import './AgenciesItemContent.scss';
```
[the remark this answers is not in the document]
Comment (Aleksandra05sh) [reply src/components/AgenciesPage/components/components/components/AgenciesItemContent/AgenciesItemContent.tsx:5]: agree with you
I don't sort imports yet

File: src/components/Agencies/Agencies.tsx (hunk 2 above)
Comment (Aleksandra05sh) [reply src/components/Agencies/Agencies.tsx]: @vovkis13 it was done in next commit
@topsart commit was renamed

File: src/components/AgenciesPage/types/Agencies.ts (hunk 3 above)
Comment (kerrimov) [reply src/components/AgenciesPage/types/Agencies.ts]: We can't use any and undefined.

File: src/components/AgenciesPage/AgenciesPage.tsx (hunk 4 above)
Comment (kerrimov) [reply src/components/AgenciesPage/AgenciesPage.tsx]: You can delete `section` if you use `Container`

File: src/components/AgenciesPage/components/components/components/AgenciesItemActions/AgenciesItemActions.tsx [hunk 10]
```diff
@@ -0,0 +1,20 @@
...
+      className="agencies-item-actions-button"
+      variant="contained"
+      size="medium"
+    >
+      <InfoOutlined
+        className="agencies-item-actions-button-icon"
+        fontSize="small"
+      />
+      Info
+    </Button>
```
Comment (kerrimov) [inline src/components/AgenciesPage/components/components/components/AgenciesItemActions/AgenciesItemActions.tsx:18]: Move this button to the shared folder.

File: src/components/AgenciesPage/components/components/components/AgenciesItemContent/AgenciesItemContent.tsx [hunk 11]
```diff
@@ -0,0 +1,30 @@
...
+
+interface AgenciesItemContentProps {
+  agency: AgencyItemContent;
+}
+
+export const AgenciesItemContent: React.FC<AgenciesItemContentProps> = ({
+  agency,
+}) => {
+  return (
```
Comment (kerrimov) [inline src/components/AgenciesPage/components/components/components/AgenciesItemContent/AgenciesItemContent.tsx:14]: you can avoid using return here

File: src/components/AgenciesPage/components/components/components/AgenciesItemContent/AgenciesItemContent.tsx [hunk 12]
```diff
@@ -0,0 +1,30 @@
...
+      <Typography
+        className="agencies-item-card-content"
+        gutterBottom
+        variant="h5"
+        component="div"
+      >
+        {agency.name}
+      </Typography>
+
+      <div className="card-details">
```
Comment (kerrimov) [inline src/components/AgenciesPage/components/components/components/AgenciesItemContent/AgenciesItemContent.tsx:25]: Why do you need `div` with `className`?

File: src/components/AgenciesPage/components/components/components/AgenciesItemContent/components/SpecificationsTable.tsx [hunk 13]
```diff
@@ -0,0 +1,40 @@
...
+
+          return (
+            specificationValue && (
+              <tr key={index}>
+                <th className="table-specification-headings">
+                  {specificationHeading}
+                </th>
+                <td className="table-specification-value">
+                  {specificationValue}
+                </td>
```
Comment (kerrimov) [inline src/components/AgenciesPage/components/components/components/AgenciesItemContent/components/SpecificationsTable.tsx:32]: Why did you decide to use native HTML instead of MUI?

File: src/components/AgenciesPage/types/Agencies.ts [hunk 14]
```diff
@@ -0,0 +1,14 @@
+export interface Agency {
+  id: number;
+  name: string;
+  type: string;
+  administrator: string;
+  country_code: string;
+  launchers: string;
+  image_url: string;
+  spacecraft?: string;
+  [key: string]: string | undefined | number;
```
Comment (kerrimov) [inline src/components/AgenciesPage/types/Agencies.ts:10]: can't use undefined

File: src/components/Router/Router.scss [hunk 15]
```diff
@@ -1,3 +1,5 @@
 .router-elements-wrapper {
-  margin-top: 4rem;
+  margin: 0 auto;
+  padding-top: 6rem;
+  max-width: 1536px;
```
Comment (kerrimov) [inline src/components/Router/Router.scss:4]: Why `1536px`?

File: src/components/AgenciesPage/types/Agencies.ts [hunk 16]
```diff
@@ -0,0 +1,14 @@
...
+}
+
+export type AgencyItemContent = Omit<Agency, 'id' | 'image_url'>;
+export type AgenciesSpecification = Omit<Agency, 'id' | 'image_url' | 'name'>;
```
Comment (kerrimov) [inline src/components/AgenciesPage/types/Agencies.ts:14]: Why do you use Omit here?

File: src/components/Router/Router.tsx [hunk 17]
```diff
@@ -5,5 +5,5 @@ import { routesConfig } from './routes';
 
 export const Router = () => {
   const routes: React.ReactElement | null = useRoutes(routesConfig);
```
Comment (kerrimov) [inline src/components/Router/Router.tsx:7]: null can't be used here. (Missed in the previous PR)

**Review by kerrimov — COMMENTED** [review]
Delete the empty file `src/components/AgenciesPage/components/AgenciesList.scss`

File: src/components/Router/Router.scss (hunk 15 above)
Comment (Aleksandra05sh) [reply src/components/Router/Router.scss:4]: because in mui it is breakpoint

File: src/components/AgenciesPage/components/components/components/AgenciesItemContent/components/SpecificationsTable.tsx (hunk 13 above)
Comment (Aleksandra05sh) [reply src/components/AgenciesPage/components/components/components/AgenciesItemContent/components/SpecificationsTable.tsx:32]: Mui table implementation is overcomplicated for our case

File: src/components/AgenciesPage/types/Agencies.ts (hunk 16 above)
Comment (Aleksandra05sh) [reply src/components/AgenciesPage/types/Agencies.ts:14]: I can use the interface and then extend it. But for me, it is more readable.
I can change it if it needed

File: src/components/AgenciesPage/AgenciesPage.tsx (hunk 4 above)
Comment (Aleksandra05sh) [reply src/components/AgenciesPage/AgenciesPage.tsx]: deleted

File: src/components/AgenciesPage/components/components/components/AgenciesItemContent/AgenciesItemContent.tsx (hunk 11 above)
Comment (Aleksandra05sh) [reply src/components/AgenciesPage/components/components/components/AgenciesItemContent/AgenciesItemContent.tsx:14]: done

File: src/components/Router/Router.scss (hunk 15 above)
Comment (vovkis13) [reply src/components/Router/Router.scss:4]: use 96rem

File: src/components/AgenciesPage/components/components/components/AgenciesItemContent/AgenciesItemContent.tsx (hunk 12 above)
Comment (Aleksandra05sh) [reply src/components/AgenciesPage/components/components/components/AgenciesItemContent/AgenciesItemContent.tsx:25]: because I need flex-container in which the table is.

File: src/components/Router/Router.scss (hunk 15 above)
Comment (Aleksandra05sh) [reply src/components/Router/Router.scss:4]: done

File: src/components/AgenciesPage/components/components/AgenciesItem.tsx (hunk 7 above)
Comment (kerrimov) [reply src/components/AgenciesPage/components/components/AgenciesItem.tsx:13]: You should leave your option and also add Dmytro's :)

File: src/components/AgenciesPage/components/components/components/AgenciesItemContent/AgenciesItemContent.tsx (hunk 12 above)
Comment (kerrimov) [reply src/components/AgenciesPage/components/components/components/AgenciesItemContent/AgenciesItemContent.tsx]: I have not found styles for `card-details`

File: src/components/AgenciesPage/components/components/AgenciesItem.tsx (hunk 7 above)
Comment (Aleksandra05sh) [reply src/components/AgenciesPage/components/components/AgenciesItem.tsx:13]: do you mean this? 

`export const AgenciesItem: React.FC<AgenciesItemProps> = ({ agency }: AgenciesItemProps)`

File: src/components/AgenciesPage/components/components/components/AgenciesItemContent/AgenciesItemContent.tsx (hunk 12 above)
Comment (Aleksandra05sh) [reply src/components/AgenciesPage/components/components/components/AgenciesItemContent/AgenciesItemContent.tsx]: sory|
looked at other class 
moved this `div`

File: src/components/Router/Router.tsx (hunk 17 above)
Comment (Aleksandra05sh) [reply src/components/Router/Router.tsx:7]: using generic is not provided by this hook

Comment (kerrimov) [reply src/components/Router/Router.tsx:7]: Please use this doc https://reactnavigation.org/docs/typescript/#annotating-useroute

File: src/components/AgenciesPage/components/components/AgenciesItem.tsx (hunk 7 above)
Comment (Aleksandra05sh) [reply src/components/AgenciesPage/components/components/AgenciesItem.tsx:13]: fixed in this way

File: src/components/Router/Router.tsx (hunk 17 above)
Comment (kerrimov) [reply src/components/Router/Router.tsx:7]: Discussed on the daily

File: src/components/AgenciesPage/types/Agencies.ts (hunk 14 above)
Comment (kerrimov) [reply src/components/AgenciesPage/types/Agencies.ts:10]: the way how fix it: https://stackoverflow.com/questions/61431397/how-to-define-typescript-type-as-a-dictionary-of-strings-but-with-one-numeric-i
